File: 11/gemini-3.0-pro/python/solution.py
import sys
from functools import lru_cache

# Increase recursion depth for deep graphs
sys.setrecursionlimit(20000)

def solve():
    try:
        with open("input.txt", "r") as f:
            lines = f.readlines()
    except FileNotFoundError:
        print("Error: input.txt not found.")
        return

    graph = {}
    for line in lines:
        line = line.strip()
        if not line:
            continue
        source, targets_str = line.split(": ")
        targets = targets_str.split(" ")
        graph[source] = targets

    # Part 1
    @lru_cache(None)
    def count_paths_p1(current):
        if current == "out":
            return 1

        if current not in graph:
            return 0

        total = 0
        for neighbor in graph[current]:
            total += count_paths_p1(neighbor)
        return total

    paths_p1 = count_paths_p1("you")
    print(f"Day 11 Part 1: {paths_p1}")

    # Part 2
    @lru_cache(None)
    def count_paths_p2(current, visited_mask):
        # Mask: 1 = dac, 2 = fft
        if current == "dac":
            visited_mask |= 1
        elif current == "fft":
            visited_mask |= 2

        if current == "out":
            return 1 if visited_mask == 3 else 0

        if current not in graph:
            return 0

        total = 0
        for neighbor in graph[current]:
            total += count_paths_p2(neighbor, visited_mask)
        return total

    paths_p2 = count_paths_p2("svr", 0)
    print(f"Day 11 Part 2: {paths_p2}")

if __name__ == "__main__":
    solve()

File: 11/gemini-3.0-pro/typescript/solution.ts
import * as fs from 'fs';

const input = fs.readFileSync('input.txt', 'utf-8');
const lines = input.split('\n');

const graph = new Map<string, string[]>();

for (const line of lines) {
    if (!line.trim()) continue;
    const [source, targetsStr] = line.trim().split(': ');
    const targets = targetsStr.split(' ');
    graph.set(source, targets);
}

// Part 1
const memo1 = new Map<string, number>();
function countPathsP1(current: string): number {
    if (current === 'out') return 1;
    if (memo1.has(current)) return memo1.get(current)!;

    if (!graph.has(current)) return 0;

    let total = 0;
    for (const neighbor of graph.get(current)!) {
        total += countPathsP1(neighbor);
    }

    memo1.set(current, total);
    return total;
}

const paths1 = countPathsP1('you');
console.log(`Day 11 Part 1: ${paths1}`);

// Part 2
const memo2 = new Map<string, number>();
function countPathsP2(current: string, mask: number): number {
    if (current === 'dac') mask |= 1;
    if (current === 'fft') mask |= 2;

    if (current === 'out') return mask === 3 ? 1 : 0;

    const key = `${current}-${mask}`;
    if (memo2.has(key)) return memo2.get(key)!;

    if (!graph.has(current)) return 0;

    let total = 0;
    for (const neighbor of graph.get(current)!) {
        total += countPathsP2(neighbor, mask);
    }

    memo2.set(key, total);
    return total;
}

const paths2 = countPathsP2('svr', 0);
console.log(`Day 11 Part 2: ${paths2}`);
